### PR TITLE
GH-4391 dynamic model namespaces

### PR DESCRIPTION
**Pull request by domkun**
GitHub issue resolved: #4391

Briefly describe the changes proposed in this PR:

Preserves namespace of the `DynamicModel` after an upgrade happened.

### CODE REVIEW

**Comment by JervenBolleman** [comment]
Dear @domkun thank you for this nice patch and test. 

I did a preliminary review and it looks nice to me. 
However, I am going to let a different RDF4j Contributor make the final call as I myself have never used these classes.

File: core/model/src/test/java/org/eclipse/rdf4j/model/impl/DynamicModelTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.model.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.model.Namespace;
+import org.junit.Test;
```
Comment (erikgb) [inline core/model/src/test/java/org/eclipse/rdf4j/model/impl/DynamicModelTest.java:18]:
```suggestion
import org.junit.jupiter.api.Test;
```

Could we use JUnit 5, please. Ref. https://github.com/eclipse/rdf4j/pull/4385

Comment (domkun) [reply core/model/src/test/java/org/eclipse/rdf4j/model/impl/DynamicModelTest.java]: Sure

**Review by abrokenjester — APPROVED** [review]
LGTM - nice fix @domkun , thanks for providing it!

**Comment by hmottestad** [comment]
Thank you @domkun for finding and fixing this issue! I'm planning a new patch release of RDF4J fairly soon and your fix will be included :)

**Comment by domkun** [comment]
Glad I could help a bit. Nice to hear that. Such a friendly conversation. Thank you all.

**Pull request merged**